### PR TITLE
Make SqlDataRecord compatible with Framework

### PR DESCRIPTION
**Pull request by saurabh500**
Adding the interface and modifying contracts for SqlDataRecord to make it compatible with Framework.

See https://github.com/dotnet/corefx/issues/17708

### CODE REVIEW

File: src/System.Data.SqlClient/ref/System.Data.SqlClient.cs [hunk 1]
```diff
@@ -116,6 +116,7 @@ public partial class SqlDataRecord
         public virtual void SetTimeSpan(int ordinal, System.TimeSpan value) { }
         public virtual void SetValue(int ordinal, object value) { }
         public virtual int SetValues(params object[] values) { throw null; }
+        System.Data.IDataReader System.Data.IDataRecord.GetData(int ordinal) { throw null; }
```
Comment (ViktorHofer) [inline src/System.Data.SqlClient/ref/System.Data.SqlClient.cs]: NIT: Methods are normally sorted alphabetically in refs and impl

**Comment by ViktorHofer** [comment]
@safern You know why this is failing?

**Comment by ViktorHofer** [comment]
@saurabh500 Please do 
`msbuild src/shims/ApiCompat.proj /t:Rebuild`
from the corefx root so that the ApiCompatBaseline.netcoreapp.netfx461.txt gets updated and the errors disappear. If you don't know how exactly I can also do that for you.

**Comment by saurabh500** [comment]
@ViktorHofer  I will take care of the baseline build.

File: src/System.Data.SqlClient/ref/System.Data.SqlClient.cs (hunk 1 above)
Comment (corivera) [inline src/System.Data.SqlClient/ref/System.Data.SqlClient.cs]: Should this have a "public" modifier?

Comment (ViktorHofer) [inline src/System.Data.SqlClient/ref/System.Data.SqlClient.cs]: The "public" modifier is implicit. I looked into other code parts in corefx and we don't have it everywhere

**Comment by stephentoub** [comment]
@dotnet-bot test Windows_NT Debug AllConfigurations Build please (fixed build break)

**Comment by ViktorHofer** [comment]
Tests are passing now. I guess you can merge after adding the baseline changes and maybe fix the order of the method.

**Comment by saurabh500** [comment]
I fixed  the order in the ref assembly.
However the baseline seems to generate a lot more diffs for other assemblies. How do I build the baseline for only SqlClient?

**Comment by ViktorHofer** [comment]
> I fixed the order in the ref assembly.

Can't see it here in the file changes.

> However the baseline seems to generate a lot more diffs for other assemblies. How do I build the baseline for only SqlClient?

I think you can't. Maybe just merge it without the baseline, I will take a look into it afterwards.

File: src/System.Data.SqlClient/ref/System.Data.SqlClient.cs (hunk 1 above)
Comment (saurabh500) [inline src/System.Data.SqlClient/ref/System.Data.SqlClient.cs]: The modifier IDataRcord will take effect here. No modifier should be used

**Comment by safern** [comment]
> I think you can't. Maybe just merge it without the baseline, I will take a look into it afterwards.

You can actually run a baseline for a single project but the BaseLine file would be added into that project itself. e.g: https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/ApiCompatBaseline.uapaot.txt

The way to do it is the following:
```
msbuild <path to csproj> /p:BaselineAllApiCompatError=true /p:TargetGroup=<the target group to baseline>
``` 

TargetGroup could be netfx, uap, uapaot, etc.

I'm not sure if this is what you guys want.

**Comment by saurabh500** [comment]
@ViktorHofer I missed a commit for alphabetical sorting.  I have added it. While running baseline for System.Data.SqlClient.csproj I get a baseline file with 0 issues for Target Group netfx. 
Can you take a look at the Baseline generation? I can merge the PR once the checks pass.

**Comment by ViktorHofer** [comment]
done

**Pull request merged**